src/main/java/seedu/address/logic/parser/UpdateCommandParser.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CANCEL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NEW;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PROCEED;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.update.UpdateCancelCommand;
+import seedu.address.logic.commands.update.UpdateCommand;
+import seedu.address.logic.commands.update.UpdateNewCommand;
+import seedu.address.logic.commands.update.UpdateProceedCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.property.status.Offer;
+
+public class UpdateCommandParser implements Parser<UpdateCommand> {
+
+    private static final String UPDATE_STRING_REGEX = "(((new)|(proceed)|(cancel)){1} (?<amount>"
+            + Offer.VALIDATION_REGEX + ")?)";
+    private static final Pattern UPDATE_STRING_FORMAT = Pattern.compile(UPDATE_STRING_REGEX,
+            Pattern.CASE_INSENSITIVE);
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the UpdateCommand
+     * and returns an UpdateCommand object for execution.
+     * @throws ParseException If the user input does not conform the expected format.
+     */
+    public UpdateCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NEW, PREFIX_PROCEED, PREFIX_CANCEL);
+
+        Matcher matcher = UPDATE_STRING_FORMAT.matcher(args);
```

Comment:
Is there supposed to be a use for this? I can't find any place that uses this `matcher` object

Reply:
I was thinking of using this to validate that there is only 1 new/proceed/cancel

Reply:
but have not written the code out fully yet

Reply:
Ohh, I see

Reply:
added the code to ensure that there is only 1 new/proceed/cancel